OrbitClientData/include/OrbitClientData/TracepointData.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_CORE_TRACEPOINT_EVENT_BUFFER_H_
+#define ORBIT_CORE_TRACEPOINT_EVENT_BUFFER_H_
+
+#include <absl/container/flat_hash_map.h>
+#include <absl/synchronization/mutex.h>
+
+#include <atomic>
+#include <functional>
+#include <limits>
+#include <map>
+#include <vector>
+
+#include "capture_data.pb.h"
+#include "tracepoint.pb.h"
+
+/*
+ * Stores `TracepointEventInfos` and a mapping from tracepoint key to unique `TracepointInfo`s.
```

Comment:
I like class comments that are understandable with very little context/knowledge. Suggestion:

TracepointData stores all tracepoint related information on the Client/Ui side. 
Single events in which a tracepoint got hit are represented by a 'TracepointEventInfo'. These events are stored ordered by the time stamp of the event such that one can iterate over the events in a time interval in an efficient manner. 
The class offers methods to add events and iterate over over them.

Other than the events themselves the class stores a list containing all the available tracepoints on the target system ('TracepointInfo's). This is stored in a map that from an integer identifier to a 'TracepointInfo'. This map is also used to compress the wire format of the events (the events contain an identifier rather than the full name of the travepoint they correspond to).